aadwarf64/aadwarf64.rst

```diff
@@ -0,0 +1,668 @@
+
+This ABI defines two CIE augmentation characters that may appear as part of
+a CIE augmentation string.
+
+1. The character 'B' indicates that associated frames are using the B key for
+   return address signing.
+
+2. The character 'G' indicates that associated frames may modify MTE tags on
+   the stack space they use.
+
+
+.. note::
+
+    1. The mark on a frame recording that it may have set MTE tags other than the
+       stack background is information which can be used when unwinding.
+
+.. raw:: pdf
+
+   PageBreak oneColumn
+
+.. _Call frame instructions:
+
+Call frame instructions (**Beta**)
+----------------------------------
+
+This ABI defines one vendor call frame instruction
+``DW_CFA_AARCH64_negate_ra_state``.
+
+.. class:: aadwarf64-vendor-cfa-operations
+
+.. table:: AArch64 vendor CFA operations
+
+   +------------------------------------+-------------+------------+-----------+-----------+
+   | Instruction                        | High 2 bits | Low 6 bits | Operand 1 | Operand 2 |
+   +====================================+=============+============+===========+===========+
+   | ``DW_CFA_AARCH64_negate_ra_state`` | 0           | ``0x2D``   | \-        | \-        |
+   +------------------------------------+-------------+------------+-----------+-----------+
+
+The ``DW_CFA_AARCH64_negate_ra_state`` operation negates bit[0] of the
+RA_SIGN_STATE pseudo-register. It does not take any operands.
+
+.. _DWARF expression operations:
+
+DWARF expression operations (**Beta**)
+--------------------------------------
+
+This ABI defines one vendor DWARF expression operation
+``DW_OP_AARCH64_operation``.
+
+.. class:: dwarf64-vendor-dwarf-expression-operations
+
+.. table:: AArch64 vendor DWARF expression operations
+
+   +-----------------------------+----------+
+   | Operation                   | Code     |
+   +=============================+==========+
+   | ``DW_OP_AARCH64_operation`` | ``0xea`` |
+   +-----------------------------+----------+
+
+The ``DW_OP_AARCH64_operation`` takes one mandatory operand encoded
+as an unsigned LEB128. Bits[6:0] of this value specify an AArch64
+DWARF Expression sub-operation. The remaining operands and the
+action performed are as specified by the sub-operation. The
+``DW_OP_AARCH64_operation`` allows this ABI to define operations
+specific to the Arm 64-bit architecture outside the encoding space
+of DWARF expression operations.
+
+.. class:: aadwarf64-dwarf-expression-sub-operations
+
+.. table:: AArch64 DWARF expression sub-operations
+
+   +-----------------------------+----------+
+   | Sub-operation               | Code     |
+   +=============================+==========+
+   | ``DW_SUB_OP_AARCH64_sign``  | ``0x00`` |
+   +-----------------------------+----------+
+
+The ``DW_SUB_OP_AARCH64_sign`` sub-operation takes a single operand
+encoded as an unsigned LEB128 operand. This value specifies a pointer
+key signing operation given in `aarch64 dwarf pointer signing operations`. The top
```

Comment:
I would either rephrase `aarch64 dwarf pointer signing operations` to `the table below` or I would make it a proper link. Currently when reading this it's unclear what it refers to.

Also, I'm not sure what the quotes do around `aarch64 dwarf pointer signing operations`. They don't do anything in the html doc in any case.

Reply:
I assume that aarch64 dwarf pointer signing operations should be in italics. if so, then it needs to be wrapped with `:title:`aarch64 dwarf pointer signing operation`` or `*aarch64 dwarf pointer signing operation*`

Reply:
`aarch64 dwarf pointer signing operation` is meant to refer to the table below. In the previous documents on developer.arm.com it is a link to the table. Just rendering it differently still keeps it vague what the term is supposed to refer to. So I would still suggest either making it a link to the table again or rephrasing it to make clear you're referring to the table.

Reply:
Made it a link and slightly modified the wording to reflect that it is referencing a table.